addons/binding/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/internal/GPIOHandlerFactory.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.gpio.internal;
+
+import static org.openhab.binding.gpio.GPIOBindingConstants.*;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.gpio.handler.GPIODigitalInputHandler;
+import org.openhab.binding.gpio.handler.GPIODigitalOutputHandler;
+import org.openhab.binding.gpio.handler.PigpioBridgeHandler;
+import org.openhab.binding.gpio.handler.PigpioLocalBridgeHandler;
+import org.openhab.binding.gpio.handler.PigpioRemoteBridgeHandler;
+
+/**
+ * The {@link GPIOHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author nils-developersinblue - Initial contribution
+ */
+public class GPIOHandlerFactory extends BaseThingHandlerFactory {
+
+    private final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = new HashSet<ThingTypeUID>();
+
+    public GPIOHandlerFactory() {
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_DIGITAL_INPUT);
```

Comment:
Is this wise, adding them to the static thing from the constructor, please check other bindings for their solutions.